src/read/read_options.rs

```diff
@@ -112,8 +126,128 @@ impl ReadOptions {
+                    }
+                    Format::Mmcif => super::mmcif::open_mmcif_raw_with_options(reader, self),
+                    Format::Auto => Err(vec![PDBError::new(
+                        crate::ErrorLevel::BreakingError,
+                        "Could not determine file type",
+                        "Could not determine the type of the gzipped file, use .pdb.gz or .cif.gz",
+                        Context::show(filename),
+                    )]),
+                }
+            } else {
+                match file_format {
+                    Format::Pdb => super::pdb::open_pdb_with_options(path, self),
+                    Format::Mmcif => super::mmcif::open_mmcif_with_options(path, self),
+                    _ => Err(vec![PDBError::new(
+                        crate::ErrorLevel::BreakingError,
+                        "Incorrect extension",
+                        "Could not determine the type of the given file extension, make it .pdb or .cif",
+                        Context::show(path.as_ref()),
+                    )])
+                }
+            }
+        } else {
+            Err(vec![PDBError::new(
+                crate::ErrorLevel::BreakingError,
+                "Missing extension",
+                "The given file does not have an extension, make it .pdb or .cif",
+                Context::show(path.as_ref()),
+            )])
+        }
+    }
+
+    /// Parse the input stream into a [`PDB`] struct. To allow for direct streaming from sources, like from RCSB.org.
+    /// Returns a PDBError if a BreakingError is found. Otherwise it returns the PDB with all errors/warnings found while parsing it.
+    ///
+    /// ## Arguments
+    /// * `input` - the input stream
+    /// * `context` - the context of the full stream, to place error messages correctly, for files this is `Context::show(filename)`.
+    /// * `options` - the options for configuring how a file/stream is parsed.
+    ///
+    /// # Related
+    /// If you want to open a file see [`crate::open_pdb`] and [`crate::open_mmcif`].
+    /// If you want to open a general file with no knowledge about the file type, see [`ReadOptions::read`].
+    pub fn read_raw<T>(&self, input: std::io::BufReader<T>) -> ReadResult
+    where
+        T: std::io::Read,
+    {
+        match self.format {
+            Format::Pdb => super::pdb::open_pdb_raw_with_options(input, Context::None, self),
+            Format::Mmcif => super::mmcif::open_mmcif_raw_with_options(input, self),
+            Format::Auto => Err(vec![PDBError::new(
+                crate::ErrorLevel::BreakingError,
+                "Could not determine file type",
+                "Could not determine the type of the input stream, set self.format",
+                Context::None,
+            )]),
+        }
+    }
+}
+
+/// Guess the file format based on the file name extensions.
+fn guess_format(filename: &str) -> Option<(Format, bool)> {
```

Comment:
@douweschulte Is this what you suggested in the previous comment? I believe now errors are only raised when `read_raw` is called directly on a stream and `options.format` is not set. In all other scenarios the file type should be inferred from the filename.

Reply:
Looks good!